product_attribute_value_combination_conditions/models/product.py

```diff
@@ -0,0 +1,113 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, api, fields, exceptions
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.constrains('attribute_line_ids')
+    def check_circular_dependency(self):
+        for attr_line in self.attribute_line_ids:
+            pred_lines = self.attribute_line_ids.filtered(
+                lambda x: attr_line.attribute_id in x.predecessors)
+            if attr_line.predecessors in pred_lines.mapped(
+                    'attribute_id'):
+                raise exceptions.Warning('There is a circular dependency')
+
+
+class ProductAttributeLine(models.Model):
+    _inherit = "product.attribute.line"
+
+    @api.one
+    @api.depends('product_tmpl_id.attribute_line_ids')
+    def _get_possible_attributes(self):
+        attrs = self.env['product.attribute']
+        for attr_line in self.product_tmpl_id.attribute_line_ids:
+            if attr_line.attribute_id != self.attribute_id:
+                attrs |= attr_line.attribute_id
+        self.possible_values = attrs.sorted()
+
+    @api.constrains('predecessors', 'initial')
+    def check_initial_has_predecessors(self):
+        if self.initial and self.predecessors:
+            raise exceptions.Warning('initial attribute cant content '
+                                     'predecessors')
+
+    @api.constrains('predecessors', 'predecessors_constraint')
+    def check_predecessors_values(self):
+        if self.predecessors_constraint - self.possible_predecessor_values:
+            raise exceptions.Warning('There are values that not longer exist')
```

Comment:
Cambiar por "There are values in constraint values that don't belong anymore to the possible ones"